### DISCUSSION TITLE
Change default bed heating parameters to read it from GCode first layer

### DISCUSSION THREAD

**Discussion by adysowski** [Guilouz/Creality-Helper-Script-Wiki#426] [Issues]
Hi, i have a question - every time I want to start printing nozzle heats up to desired first layer temperature, bed also, but then it suddenly comes to 170*C nozzle and 50*C on bed, then starts meshing. I found these parameters to be in printer_params.cfg, why does it behave like this? When printing ABS on 100*C bed the 50*C mesh is useless, because "warps" are changing dynamically with temperature. I use OrcaSlicer and except that everything works perfect and smooth. Thanks :) Maybe I have something wrong in setup? My start g-code is:
`M190 S[bed_temperature_initial_layer_single]
M109 S[nozzle_temperature_initial_layer]
START_PRINT EXTRUDER=[nozzle_temperature_initial_layer] BED=[bed_temperature_initial_layer_single]
`

**Comment by Guilouz** [comment]
You must use this start gcode:

```
SET_PRINT_STATS_INFO TOTAL_LAYER=[total_layer_count]
M140 S0
M104 S0
START_PRINT EXTRUDER_TEMP=[nozzle_temperature_initial_layer] BED_TEMP=[bed_temperature_initial_layer_single]
```

**Reply by adysowski** [reply]
Yeah, it works brilliantly! Thank you very much :) U are doing great job, i love my k1 max mainly cause of Your script ;") You can add new section in Wiki about this G-Code ;) Thanks once again.
